File: leetsync/2232-adding-spaces-to-a-string/adding-spaces-to-a-string.go
func addSpaces(s string, spaces []int) string {
	result := make([]uint8, 0, len(s)+len(spaces))

	for sIterator, spaceIterator := 0, 0; sIterator < len(s); sIterator++ {
		if spaceIterator < len(spaces) && sIterator == spaces[spaceIterator] {
			result = append(result, ' ')
			spaceIterator++
		}
		result = append(result, s[sIterator])
	}
	return string(result)
}

File: leetsync/2232-adding-spaces-to-a-string/adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder sb = new StringBuilder();
        int m = s.length();
        int n = spaces.length;
        int j = 0;
        for (int i = 0; i < m; i++) {
            if (j < n && i == spaces[j]) {
                sb.append(" ");
                j++;
            }
            sb.append(s.charAt(i));
        }

        return sb.toString();
    }
}